dev/sparktestsupport/toposort.py

```diff
@@ -0,0 +1,85 @@
+#######################################################################
+# Implements a topological sort algorithm.
```

Comment:
Do I just append the entire contents of the notice verbatim? Is there a shorthand way of doing this (e.g. just including just the copyright)? I got a bit confused looking at the existing NOTICE file and since it hasn't been updated in a while I didn't spot any smaller patches / diffs to model my change on.